test/unit/rounding/fixed_point/elastic_integer/elastic_fixed_point.cpp

```diff
@@ -112,4 +112,81 @@ namespace {
                 -10),
                 "");
     }
+
+    namespace efxp_round_test{
+        using T0 = cnl::elastic_fixed_point<24, -20>;
+        using T1 = cnl::elastic_fixed_point<16, -8>;
+        using T2 = cnl::elastic_fixed_point<8, -4>;
+        using T3 = cnl::elastic_fixed_point<5, -1>;
+    }
+
+    namespace elastic_fixed_point_implicit_conversions {
+        using efxp_round_test::T1;
+        using efxp_round_test::T2;
+        using efxp_round_test::T3;
+
+        static_assert(identical(T2{0.25}, static_cast<T2>(T1{0.25})),
```

Comment:
A good test, is simple and self-contained. You should be able to look at it as a form of documentation and it should tell you something about the library.

These tests have meaningless names for types: `T0`, `T1` etc., and they say self-evident things: if you have a value, `v`, and you cast it to type `T` then you have a value `v` of type `T`. That says nothing very helpful. Even if it involves some repetition, a test that you can look at and understand what it's doing is really valuable:

```c++
static_assert(
    identical(
        cnl::elastic_fixed_point<8, -4>{0.25},
        static_cast<cnl::elastic_fixed_point<8, -4>>(cnl::elastic_fixed_point<16, -8>{0.25})),
```

When it is spelled out like this, I can see that the tests which don't use `T3` probably aren't that useful because it's only when the conversion is to something that cannot represent quarter values that the rounding will have an effect. Otherwise, the type has no problem storing the value.

Reply:
In terms of testing I tend to be paranoid, thus test even the stupid ones, but you are absolutely correct. Those additional tests do not provide additional value. I will take another look and make another PR for correcting these.